drivers/clock_control/clock_control_mcux_scg.c

```diff
@@ -60,33 +61,56 @@ static int mcux_scg_get_rate(const struct device *dev,
 	case KINETIS_SCG_FIRC_CLK:
 		clock_name = kCLOCK_ScgFircClk;
 		break;
+#if (FSL_FEATURE_SCG_HAS_SPLL)
 	case KINETIS_SCG_SPLL_CLK:
 		clock_name = kCLOCK_ScgSysPllClk;
 		break;
+#endif
+#if (FSL_FEATURE_SCG_HAS_LPFLL)
+	case KINETIS_SCG_SPLL_CLK:
+		clock_name = kCLOCK_ScgLpFllClk;
+		break;
+#endif
+#if (FSL_FEATURE_SCG_HAS_SOSCDIV1)
 	case KINETIS_SCG_SOSC_ASYNC_DIV1_CLK:
 		clock_name = kCLOCK_ScgSysOscAsyncDiv1Clk;
 		break;
+#endif
 	case KINETIS_SCG_SOSC_ASYNC_DIV2_CLK:
 		clock_name = kCLOCK_ScgSysOscAsyncDiv2Clk;
 		break;
+#if (FSL_FEATURE_SCG_HAS_SIRCDIV1)
 	case KINETIS_SCG_SIRC_ASYNC_DIV1_CLK:
 		clock_name = kCLOCK_ScgSircAsyncDiv1Clk;
 		break;
+#endif
 	case KINETIS_SCG_SIRC_ASYNC_DIV2_CLK:
 		clock_name = kCLOCK_ScgSircAsyncDiv2Clk;
 		break;
+#if (FSL_FEATURE_SCG_HAS_FIRCDIV1)
```

Comment:
why did all the #if come back

Reply:
Intended, currently would be a better solution for the KE parts

Reply:
is it because these enums dont exist on ke15?

Reply:
The initial version of this file was fairly clean but the KE15 it didn't align. The internal team decied to leverage the FSL_FEATURE defines to wrap them which should keep this file safe for the additional parts.